### PR TITLE
IBX-7236: Suggestions only work with entire words

### PR DESCRIPTION
**Pull request by kisztof**
| Question      | Answer
| ------------- | ---
| Tickets       |  [IBX-7236](https://issues.ibexa.co/browse/IBX-7236) 
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ibexa/search/blob/main/LICENSE)

This PR fixes:
- an issue with min_query_lenght settings
- an issue with finding only entire words
- related PR https://github.com/ibexa/core/pull/298

#### Checklist:
- [ ] Implement tests
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

**Review by Steveb-p — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]
Apart from failing CI.

**Review by adamwojs — CHANGES_REQUESTED** [review]
Could you please add test case for a fixed issue and add meaningful PR title ?

File: src/bundle/Controller/SuggestionController.php [hunk 1]
```diff
@@ -17,14 +18,23 @@ final class SuggestionController extends AbstractController
...
+        ConfigResolverInterface $configResolver
     ) {
         $this->suggestionService = $suggestionService;
+        $this->configResolver = $configResolver;
     }
 
     public function suggestAction(SuggestionQuery $suggestionQuery): JsonResponse
     {
+        $minQueryLength = $this->configResolver->getParameter('search.suggestion.min_query_length');
+        if (strlen($suggestionQuery->getQuery()) < $minQueryLength) {
```
[the remark this answers is not in the document]
Comment (kisztof) [reply src/bundle/Controller/SuggestionController.php]: Absolutely agree

**Comment by webhdx** [comment]
@kisztof Can you fix PHPStan errors? You only need to remove some ignored entries.

**Review by webhdx — APPROVED** [review]

**Comment by bogusez** [comment]
@kisztof here you have several examples where search suggestions still does not work correctly and one which works good (I have added screenshot newFolder_correct to give you comparison that some phrases are treated correctly by search engine in terms of suggestions).

Incorrect behaviour:

Screenshot 2023-11-29 at 13 04 18
Screenshot 2023-11-29 at 13 17 36
Screenshot 2023-11-29 at 13 19 10
Screenshot 2023-11-29 at 13 20 32

Correct behaviour:

newFolder_correct

**Comment by webhdx** [comment]
I think you should wrap the query with wildcard character on both sides. Righty now, you essentially search for content items where the name is prefixed with the query.

**Comment by adamwojs** [comment]
@kisztof Could you please address my comment https://github.com/ibexa/search/pull/37#pullrequestreview-1754985613 ? 

> Could you please add test case for a fixed issue and add meaningful PR title ?

**Review by Nattfarinn — APPROVED** [review]

File: src/bundle/Controller/SuggestionController.php [hunk 2]
```diff
@@ -17,14 +18,23 @@ final class SuggestionController extends AbstractController
...
+        ConfigResolverInterface $configResolver
     ) {
         $this->suggestionService = $suggestionService;
+        $this->configResolver = $configResolver;
     }
 
     public function suggestAction(SuggestionQuery $suggestionQuery): JsonResponse
     {
+        $minQueryLength = $this->configResolver->getParameter('search.suggestion.min_query_length');
+        if (mb_strlen($suggestionQuery->getQuery()) < $minQueryLength) {
```
Comment (mikadamczyk) [inline src/bundle/Controller/SuggestionController.php:34]: just to be sure, `$suggestionQuery->getQuery()) < $minQueryLength` or `$suggestionQuery->getQuery()) <= $minQueryLength`?

File: src/lib/View/SearchViewFilter.php [hunk 3]
```diff
@@ -78,6 +78,7 @@ public function handleSearchForm(FilterViewBuilderParametersEvent $event): void
 
         $request = $event->getRequest();
 
+        /** @var array<string, mixed> $search */
```
Comment (mikadamczyk) [inline src/lib/View/SearchViewFilter.php:81]: something strange happened here. This was resolved differently in `https://github.com/ibexa/search/pull/40/files#diff-8bad468f5d65de5f0490ca72ba5de084d0ddfa85f17b56fba1a73a01a64d5777R81` and I think that was better.

**Review by mikadamczyk — CHANGES_REQUESTED** [review]

File: src/bundle/Controller/SuggestionController.php (hunk 2 above)
Comment (webhdx) [reply src/bundle/Controller/SuggestionController.php:34]: It's fine, the condition is reversed - quit if query length is lower than configured limit.

**Review by bogusez — APPROVED** [review]
Regression tests passed

**Pull request merged**